#### 1. Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean):

-  среднее арифметическое
-  среднее квадратичное отклонение
-  смещенную и несмещенную оценки дисперсий для данной выборки.

In [3]:
import numpy as np
import matplotlib.pyplot as plt
from math import factorial

In [4]:
arr=np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])

а) Найти среднее арифметическое, для этого используем формулу: $\displaystyle M(X)=\frac{1}{n}\sum^n_{i=1}x_i$

In [5]:
def mean_value(array):
    return sum(array) / len(array)

In [6]:
print(f'Среднее арифметическое = {round(mean_value(arr),2)}')
a=round(mean_value(arr),2)

Среднее арифметическое = 65.3


In [7]:
np.mean(arr)

65.3

б) Расчитать среднее квадратичное отклонение, для этого используем формулу: $\sigma=\sqrt{\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n-1}}$ , где $\bar x =\frac{1}{n}\sum^n_{i=1}x_i$

In [8]:
def mean_square_deviation(array):
    square_deviation = (array - mean_value(array)) ** 2
    return (sum(square_deviation) / (len(square_deviation) - 1)) ** (1/2)

In [11]:
print(f'Среднее квадратичное отклонение = {round(mean_square_deviation(arr),2)}')

Среднее квадратичное отклонение = 31.62


In [10]:
np.std(arr, ddof=1)

31.624607341019814

в) Рассчитать смещенную и несмещенную оценки дисперсий, для этого используем формулы:

смещенной дисперсии: $\sigma^2=\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n}$  
несмещённой дисперсии: $\sigma^2=\frac{\displaystyle\sum^n_{i=1}(x_{i}-\bar x)^2}{n-1}$

In [38]:
def dispersion(array, offset):
    square_deviation = (array - mean_value(array)) **2
    return sum(square_deviation) / (len(square_deviation) - 1) if offset else sum(square_deviation) / len(square_deviation)

In [39]:
print(f'Смещенная оценка дисперсии = {round(dispersion(arr, False),2)}\n'
      f'Немещенная оценка дисперсии = {round(dispersion(arr, True),2)}')

Смещенная оценка дисперсии = 950.11
Немещенная оценка дисперсии = 1000.12


In [37]:
np.var(arr, ddof=0)

950.11

In [36]:
np.var(arr, ddof=1)

1000.1157894736842

### Результат:  
а) Среднее арифметическое $М(Х) = 65.30 $;  
б) Среднее квадратичное отклонение $\sigma = 31.62$;  
в) Смещенная оценка дисперсии $\sigma^2 = 950.11$  
   Немещенная оценка дисперсии $\sigma^2 = 1000.12$